feat: fetch and import matching training data commands (#1233)
Nix-Security-WG/nix-security-tracker: docs/data_ingestion_and_matching.md

```

This should produce untriaged matches.

### Offline matching training data

> To reproduce curated matching labels locally (for offline algorithm benchmarks), fetch the bounded JSON corpus from staging or production and import it onto a synthetic `benchmark` channel. You need a Knox API token for a user in the `matching_training_data` group.

Reviewer: ```suggestion
In order to measure matching algorithm accuracy locally when developing, you will need access to production data, which contains past automatic matches curated with corrections by users. 
You need a [API token](https://tracker.security.nixos.org/ui-v2/user/tokens), and permissions to access the endpoint with your user account which you can request by contacting maintainers.
```

> 1. Fetch paginated pages (about 7–10 MiB / ~8k records on staging):

Reviewer: ```suggestion
1. Fetch paginated pages (ca. 500k items total):
```

> 

Reviewer: Indent the list item body, otherwise you'll just have free-floating paragraphs that happen to start with numbers.

```console

> export MATCHING_TRAINING_DATA_TOKEN=<knox-token>

Reviewer: ```suggestion
export MATCHING_TRAINING_DATA_TOKEN=<api-token>
```

manage fetch_matching_training_data \
  --base-url https://tracker.security.nixos.org \
  --output ./training-data/

> ```

Reviewer: ````suggestion
```

This will produce one JSON file per page fetched.
````

> 2. Import into your local database (do **not** run `manage listen` during import; the command suppresses pgpubsub triggers so cache/linkage workers are not flooded):

Reviewer: ```suggestion
2. Import into your local database

   > [!WARNING]: Do **not** run `manage listen` during import.
   > The command suppresses `pgpubsub` triggers so workers are not flooded with tasks.
```

```console
manage import_matching_training_data --input ./training-data/
```

> Re-importing the same dump is idempotent per CVE id. A follow-up `benchmark_matching` command will score candidate matchers against these labels.

Reviewer: ```suggestion
Re-importing the same dump is idempotent per CVE ID.
```
